src/mc-fle2-payload-iev-private-v2.h

```diff
@@ -122,12 +133,24 @@ bool mc_FLE2IndexedEqualityEncryptedValueV2_parse(mc_FLE2IndexedEncryptedValueV2
  * 1/ `edge_count` is introduced as an octet following `original_bson_type`.
  * 2/ Rather than a single metadata block, we have {edge_count} blocks.
  *
- * Since libmongocrypt ignores metadata blocks, we can ignore most all
- * differences between Equality and Range types for IndexedEncrypted data.
  */
 
 bool mc_FLE2IndexedRangeEncryptedValueV2_parse(mc_FLE2IndexedEncryptedValueV2_t *iev,
                                                const _mongocrypt_buffer_t *buf,
                                                mongocrypt_status_t *status);
 
+/*
+ * Serializes an mc_FLE2IndexedEncryptedValueV2_t into a buffer.
+ *
+ * The serialized output follows the same layout as the input `buf` to
+ * mc_FLE2IndexedRangeEncryptedValueV2_parse, allowing for round-trip
+ * conversions between the serialized and parsed forms.
+ *
+ * Retuns an error if the input structure is not valid, or if the buffer
+ * provided is insufficient to hold the serialized data.
+ */
+bool mc_FLE2IndexedRangeEncryptedValueV2_serialize(const mc_FLE2IndexedEncryptedValueV2_t *iev,
```

Comment:
Let's add `mc_FLE2IndexedEqualityEncrytedValueV2_serialize()` while we're here.

Reply:
Addressed with the consolidation of functions.

Does this address [MONGOCRYPT-608](https://jira.mongodb.org/browse/MONGOCRYPT-608)? I'm not certain because the `_write` function in v1 takes each individual `mc_FLE2IndexedEncryptedValue_t` field and serializes it to a buffer (vs. the v2 `_serialize` function which does the same but from a `mc_FLE2IndexedEncryptedValueV2_t` struct passed through to the function).

Functionally, these two methods do the same thing but the `_serialize` function requires the user to create a `mc_FLE2IndexedEncryptedValueV2_t` object before calling the function. I'm unsure on what will be easiest from the `SERVER` perspective and will defer to you.

Honestly, the `_write` function implementation might be as simple as creating a `mc_FLE2IndexedEncryptedValueV2_t` object with the parameters and then calling `_serialize` (?).

Reply:
I thought about it overnight and a good checksum on these tickets might be to bring latest libmongocrypt into your local SERVER checkout (along with this change) and see if that enables implementing SERVER-95887 and SERVER-95889.